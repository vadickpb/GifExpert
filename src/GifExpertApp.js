import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {

    //const categories = ['goku', 'naruto', 'saint seia'];
    const [categories, setCategories] = useState(['goku']);

    // const handleAdd = () => {
    //     //setCategories([...categories, 'HunterX']);
    //     setCategories( cats => [...cats, 'HunderXHunter']);
    // }

    
    return (
        <>
            <h2>Giff Expert App</h2>
            <AddCategory setCategories = { setCategories}/>

            <hr />

            {/* <button onClick = {handleAdd}>Agregar</button> */}
            <ol>
                {
                    categories.map(category => 
                    <GifGrid 
                        key ={category}
                        category = {category}
                        />
                    )
                }
            </ol>
        </>
    )
}

export default GifExpertApp;