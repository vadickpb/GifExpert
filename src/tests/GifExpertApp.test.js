import { shallow } from 'enzyme';
import React from 'react';
import GifExpertApp from '../GifExpertApp';

import '@testing-library/jest-dom'

describe('Pruebas en <GifExpertApp />>', () => {
    
    test('debe mostrar el contenido', () => {
        
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot()

    })

    test('Debe mostrar una lista de categorias', () => {
        const defaultCategories = ['One Punch', 'Goku'];
        const wrapper = shallow(<GifExpertApp defaultCategories = {defaultCategories} />)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(defaultCategories.length )
    })
    
    
})
