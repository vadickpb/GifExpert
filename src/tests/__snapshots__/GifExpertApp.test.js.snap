// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp />> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Giff Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp />> debe mostrar el contenido 1`] = `
<Fragment>
  <h2>
    Giff Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
